compiler/src/iree/compiler/ConstEval/JitGlobals.cpp

```diff
@@ -89,6 +90,240 @@ class SupportedFeatures {
+      return Availability::Unknown;
+    return it->second;
+  }
+
+private:
+  void run(Operation *rootOp, SymbolTable &symbolTable,
+           const IREE::Util::ConstExprAnalysis &constExprAnalysis) {
+    unsigned nextOpOrdinal = 0;
+    for (auto &region : rootOp->getRegions()) {
+      for (auto &op : region.getOps()) {
+        if (auto globalOp = dyn_cast<IREE::Util::GlobalOpInterface>(op)) {
+          // Globals with initial values are initialized in order with where
+          // they are in the module.
+          auto &timeline = globalTimelines[globalOp.getGlobalName().getValue()];
+          assert(timeline.empty() && "out-of-order global store");
+          timeline.push_back(
+              std::make_pair(nextOpOrdinal++, Availability::Compiler));
+        } else if (auto initializerOp =
+                       dyn_cast<IREE::Util::InitializerOpInterface>(op)) {
+          // Initializer availability depends on all dependent initialized
+          // values.
+          initializerAvailability[initializerOp] =
+              calculateInitializerAvailability(
+                  initializerOp, symbolTable, constExprAnalysis, nextOpOrdinal);
+        }
+      }
+    }
+  }
+
+  // Returns the availability of |globalName| by the time |opOrdinal| is
+  // executed. Note that some globals may be initialized multiple times (yuck,
+  // but valid).
+  Availability queryGlobalInitializationStatus(StringRef globalName,
+                                               unsigned opOrdinal) {
+    auto &timeline = globalTimelines[globalName];
+    if (timeline.empty())
+      return Availability::Unknown;
+    for (auto &timepoint : timeline) {
+      if (timepoint.first > opOrdinal)
+        return timepoint.second;
+    }
+    return timeline.back().second;
+  }
+
+  // Returns true if the given |initializerOp| is a constant expression that is
+  // able to be evaluated by this pass.
+  Availability calculateInitializerAvailability(
+      IREE::Util::InitializerOpInterface initializerOp,
+      SymbolTable &symbolTable,
+      const IREE::Util::ConstExprAnalysis &constExprAnalysis,
+      unsigned &nextOpOrdinal) {
+    SmallVector<std::pair<IREE::Util::GlobalStoreOpInterface, unsigned>>
+        globalStoreOps;
+
+    // Assume compile-time availability unless we see anything that may prevent
+    // it. As we analyze the initializer we may "lower" the availability from
+    // the most available (compile-time) to least available (run-time/unknown).
+    auto availability = Availability::Compiler;
+    auto lowerAvailability = [&](Availability newAvailability,
+                                 StringRef reason) {
+      auto previousAvailability = availability;
+      availability = static_cast<Availability>(
+          std::min(static_cast<unsigned>(availability),
+                   static_cast<unsigned>(newAvailability)));
+      if (previousAvailability != availability)
+        emitWarning(initializerOp.getLoc()) << reason;
```

Comment:
I see a bunch of warnings coming from this with the SDXL model in our test suite. Should we drop the verbosity on this logging down, or are these useful to keep in?

```
____________ IREE compile and run: sdxl-scheduled-unet-3-tank::gpu_vulkan_real_weights ____________ Error invoking iree-compile
Error code: 1
Stderr diagnostics:
Stderr diagnostics:
/home/sai/SHARK-Turbine/stable_diffusion_xl_base_1_0_PNDM_64_1024x1024_fp16_unet_3.mlir:1740:10: warning: skipping consteval initializer: has non-const-expr values
/home/sai/SHARK-Turbine/stable_diffusion_xl_base_1_0_PNDM_64_1024x1024_fp16_unet_3.mlir:1717:10: note: called from
/home/sai/SHARK-Turbine/stable_diffusion_xl_base_1_0_PNDM_64_1024x1024_fp16_unet_3.mlir:1743:11: warning: skipping consteval initializer: has runtime-dependent global load
/home/sai/SHARK-Turbine/stable_diffusion_xl_base_1_0_PNDM_64_1024x1024_fp16_unet_3.mlir:1717:10: note: called from
/home/sai/SHARK-Turbine/stable_diffusion_xl_base_1_0_PNDM_64_1024x1024_fp16_unet_3.mlir:1801:11: warning: skipping consteval initializer: uses parameters or other runtime-dependent values
/home/sai/SHARK-Turbine/stable_diffusion_xl_base_1_0_PNDM_64_1024x1024_fp16_unet_3.mlir:1717:10: note: called from
/home/sai/SHARK-Turbine/stable_diffusion_xl_base_1_0_PNDM_64_1024x1024_fp16_unet_3.mlir:1824:11: warning: skipping consteval initializer: uses parameters or other runtime-dependent values
/home/sai/SHARK-Turbine/stable_diffusion_xl_base_1_0_PNDM_64_1024x1024_fp16_unet_3.mlir:1717:10: note: called from
/home/sai/SHARK-Turbine/stable_diffusion_xl_base_1_0_PNDM_64_1024x1024_fp16_unet_3.mlir:1849:11: warning: skipping consteval initializer: has non-const-expr values
/home/sai/SHARK-Turbine/stable_diffusion_xl_base_1_0_PNDM_64_1024x1024_fp16_unet_3.mlir:1717:10: note: called from
/home/sai/SHARK-Turbine/stable_diffusion_xl_base_1_0_PNDM_64_1024x1024_fp16_unet_3.mlir:1852:11: warning: skipping consteval initializer: has runtime-dependent global load
/home/sai/SHARK-Turbine/stable_diffusion_xl_base_1_0_PNDM_64_1024x1024_fp16_unet_3.mlir:1717:10: note: called from
/home/sai/SHARK-Turbine/stable_diffusion_xl_base_1_0_PNDM_64_1024x1024_fp16_unet_3.mlir:1917:11: warning: skipping consteval initializer: uses parameters or other runtime-dependent values
/home/sai/SHARK-Turbine/stable_diffusion_xl_base_1_0_PNDM_64_1024x1024_fp16_unet_3.mlir:1717:10: note: called from
/home/sai/SHARK-Turbine/stable_diffusion_xl_base_1_0_PNDM_64_1024x1024_fp16_unet_3.mlir:1940:11: warning: skipping consteval initializer: uses parameters or other runtime-dependent values
```

Reply:
https://github.com/iree-org/iree/pull/17198 will quiet the warnings